src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala

```diff
@@ -80,26 +80,56 @@ object AnalyzerContext {
       forAnalyzers: Seq[Analyzer[_, Metric[_]]])
     : Seq[SimpleMetricOutput] = {
 
-    val selectedMetrics = analyzerContext.metricMap
+    analyzerContext.metricMap
+      // Get matching analyzers
       .filterKeys(analyzer => forAnalyzers.isEmpty || forAnalyzers.contains(analyzer))
-      .values
+      // Get analyzers with successful results
+      .filter({ case (_, metrics) =>
+      metrics.value.isSuccess })
+      // Get metrics as Double and replace simple name with description
+      .flatMap({ case (analyzer, metrics) =>
+      metrics.flatten().map(renameMetric(_, describeAnalyzer(analyzer))) })
+      // Simplify metrics
+      .map(SimpleMetricOutput(_))
       .toSeq
+  }
 
-    val metricsList = selectedMetrics
-      .filter(_.value.isSuccess) // Get analyzers with successful results
-      .flatMap(_.flatten()) // Get metrics as double
-      .map { doubleMetric =>
-        SimpleMetricOutput(
-          doubleMetric.entity.toString,
-          doubleMetric.instance,
-          doubleMetric.name,
-          doubleMetric.value.get
-        )
-      }
-
-    metricsList
+  private[this] def renameMetric(metric: DoubleMetric, newName: String): DoubleMetric = {
+    metric.copy(name = newName)
   }
 
-  private[this] case class SimpleMetricOutput(entity: String, instance: String, name: String,
-    value: Double)
+  /**
+    * Describe the Analyzer, using the name of the class, including the value of `where` field
+    * if exists.
+    * It helps us to show more readable success metrics
+    * (see https://github.com/awslabs/deequ/issues/177 for details)
+    *
+    * @param analyzer the Analyzer to be described
+    * @return the description of the Analyzer
+    */
+  private[this] def describeAnalyzer(analyzer: Any): String = {
+    val name = analyzer.getClass.getSimpleName
+
+    ReflectionUtils.getFieldValue(analyzer, "where") match {
```

Comment:
I dont think we should do the check whether the analyzer supports filtering via reflection. This is brittle and the compiler cannot check cases where we would for example rename the field internally. Please add an extra trait to mark analyzers that have a where clause and add a method to retrieve the filter. We can pattern match on the trait then here.

Reply:
Makes sense, replaced reflection with `FilterableAnalyzer` trait and added to the analyzers